### PR TITLE
app-text/uudeview: fix build for clang16

### PR DESCRIPTION
**Pull request by Schievel1**
Closes: https://bugs.gentoo.org/874960
Signed-off-by: Pascal Jäger <pascal.jaeger@leimstift.de>

@hannob

### CODE REVIEW

File: app-text/uudeview/files/uudeview-0.5.20-fix-build-for-clang16.patch [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+ 	char	*cp;
+@@ -330,9 +335,8 @@ gen_frompath()
+  *	Side effects:	None.
+  */
+ 
+-strneql(a, b, n)
+-register char *a, *b;
+-int	n;
++int 
++strneql (register char *a, register char *b, int n)
```
Comment (thesamesam) [inline app-text/uudeview/files/uudeview-0.5.20-fix-build-for-clang16.patch:196]: Please drop register whenever you see it, if you're already touching the line. It does nothing and hopefully it'll be removed from C one day.

File: app-text/uudeview/uudeview-0.5.20-r3.ebuild [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (thesamesam) [inline app-text/uudeview/uudeview-0.5.20-r3.ebuild:4]: 8, while we're here

File: app-text/uudeview/files/uudeview-0.5.20-fix-build-for-clang16.patch [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+Pascal Jäger <pascal.jaeger@leimstift.de> (2022-11-24)
+
+--- a/inews/clientlib.c
++++ b/inews/clientlib.c
+@@ -14,6 +14,9 @@ static char	*sccsid = "@(#)clientlib.c	1.11	(Berkeley) 10/27/89";
+ #include "../config.h"
+ #endif
+ 
++#define _DEFAULT_SOURCE
```
Comment (thesamesam) [inline app-text/uudeview/files/uudeview-0.5.20-fix-build-for-clang16.patch:15]: What was this for exactly?

Comment (Schievel1) [reply app-text/uudeview/files/uudeview-0.5.20-fix-build-for-clang16.patch:15]: Inet_ntoa() needs that

File: app-text/uudeview/files/uudeview-0.5.20-fix-build-for-clang16.patch (hunk 1 above)
Comment (Schievel1) [reply app-text/uudeview/files/uudeview-0.5.20-fix-build-for-clang16.patch:196]: Let's see if we can automate that.

**Comment by hannob** [comment]
I'm wondering if we should avoid doing too much here.
It seems you're both fixing the missing declarations and doing a whole lot of refactoring from K&R C to ANSI C.

I will likely fork uudeview and cleanup the whole codebase (including removing deprecated features like the register keyword and K&R declarations), but I'd prefer for now if we'd stick to a small patch adressing the issue of undefined function declarations.

**Comment by Schievel1** [comment]
> I'm wondering if we should avoid doing too much here. It seems you're both fixing the missing declarations and doing a whole lot of refactoring from K&R C to ANSI C.
> 
> I will likely fork uudeview and cleanup the whole codebase (including removing deprecated features like the register keyword and K&R declarations), but I'd prefer for now if we'd stick to a small patch adressing the issue of undefined function declarations.

Sure, I can remove the K&R to ANSI things from the patch. I usually include them, because this is throwing warnings and with cproto they are a very low hanging fruit.

File: app-text/uudeview/files/uudeview-0.5.20-fix-build-for-clang16.patch (hunk 1 above)
Comment (Schievel1) [reply app-text/uudeview/files/uudeview-0.5.20-fix-build-for-clang16.patch]: hanno will do it later himselv

**Comment by Schievel1** [comment]
alright, I rewrote the patch so it only does the things that are needed for it to build with clang16.

**Comment by hannob** [comment]
gen_frompath does not return anything, so it can be void. otherwise this looks ok, I'll commit together with the format string fix and this change.

**Pull request closed without merging**